### ISSUE TITLE
Indent without sorting of attributes

### ISSUE THREAD

**Issue by Jones-S** [alek-sys/sublimetext_indentxml#59]
Hey there
I'm not sure whether this is the right place to address this issue, but I did not find any comment function on github.
Thanks for this plugin first of all! Helps me a lot.
there is just one thing: I don't want the plugin to rearrange my attributes of my xml nodes.
Could you tell me where I can disable that feature?
Thx a lot

Jonas

**Comment by iGemini** [comment]
I'd also greatly appreciate the ability to indent my XML without the attributes being re-arranged.

**Comment by giovannibgr** [comment]
+1 - sorting the attributes adds a lot of difficulty when trying to diff the previous (unformatted) and the formatted version of an XML file.
However, I'm not very optimistic if this can be fixed, it may be inherent to minidom, which this plugin uses.

**Comment by raghu777** [comment]
+1 - Need this feature.

**Comment by mgruel** [comment]
+1

**Comment by vclayton** [comment]
For those looking to indent JSON without sorting, you can edit your local copy of the package. Change the last line from:
`return json.dumps(parsed, sort_keys=True, indent=4, separators=(',', ': '), ensure_ascii=False)`
to
`return json.dumps(parsed, sort_keys=False, indent=4, separators=(',', ': '), ensure_ascii=False)`

**Comment by orchetect** [comment]
+1

Also a config parameter for number of spaces for each indent would be nice.

**Comment by cederlof** [comment]
+1

**Comment by slishnevsky** [comment]
+100500, please don't change the order of attributes

**Comment by hualig** [comment]
I am using on Mac, where can I find local copy of package? It seems package is unzipped and no way to open it.

**Comment by fsinisi90** [comment]
> For those looking to indent JSON without sorting, you can edit your local copy of the package. Change the last line from:
> `return json.dumps(parsed, sort_keys=True, indent=4, separators=(',', ': '), ensure_ascii=False)`
> to
> `return json.dumps(parsed, sort_keys=False, indent=4, separators=(',', ': '), ensure_ascii=False)`

It didn't work for me. I've even created a fork with that line changed, installed my own package and it's still reordering the attributes.

**Comment by fsinisi90** [comment]
> +1 - sorting the attributes adds a lot of difficulty when trying to diff the previous (unformatted) and the formatted version of an XML file.
> However, I'm not very optimistic if this can be fixed, it may be inherent to minidom, which this plugin uses.

You're right, another tool should be used according to [this stackoverflow answer](https://stackoverflow.com/a/662713/4644486).

**Comment by panda7789** [comment]
Or can you please at least write it to the readme file? 🙏
